### PR TITLE
Reader Stream Refresh: add gallery card

### LINKED ISSUE #8520: Reader Stream Refresh: add card with a photo gallery
For posts with several usable photos, add a new photo gallery card type:

card-example-gallery

For the cards at all sizes, see https://cloudup.com/c-DVwphUzIi.
- [ ] Build HTML/CSS (@jancavan)
- [ ] Integrate with real post data

### PR DESCRIPTION
**Pull request by bluefuton**
Fixes #8520.

8d5adfd2-8bc4-11e6-89c9-e4d0276cfe85

A feed with several gallery posts: http://calypso.localhost:3000/read/feeds/15197253

### CODE REVIEW

**Comment by bluefuton** [comment]
@jancavan logic to display the first four images on the gallery card is in place, with very minimal styling:

screen shot 2016-10-20 at 16 30 57

**Comment by bluefuton** [comment]
### todo

- [ ] Style gallery cards (@jancavan)
- [ ] Add image alt/title (@bluefuton)

**Comment by bluefuton** [comment]
I've attempted to add `alt` to the images, but it's getting stripped back off in a later render:

screen shot 2016-10-31 at 17 29 32

File: client/lib/post-normalizer/rule-wait-for-images-to-load.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ import debugFactory from 'debug';
 const debug = debugFactory( 'calypso:post-normalizer:wait-for-images-to-load' );
 
 function convertImageToObject( image ) {
-	return pick( image, [ 'src', 'naturalWidth', 'naturalHeight' ] );
+	return pick( image, [ 'src', 'naturalWidth', 'naturalHeight', 'alt' ] );
```
Comment (blowery) [inline client/lib/post-normalizer/rule-wait-for-images-to-load.js]: this is the `Image` object from `imageForUrl`, not the `img` tag in the dom. You'll need to alter `imageForUrl` to pass along the `alt` (and probably the `title` too?) and set those on the image object to pull 'em back off.

**Comment by jancavan** [comment]
@bluefuton We're going to have to use a background image here since all images within the gallery have a width of 25% the entire container: https://github.com/Automattic/wp-calypso/issues/8520#issuecomment-251834515

**Comment by jancavan** [comment]
Fixed gallery images not displaying in: https://github.com/Automattic/wp-calypso/pull/8872/commits/0609d1966b0d94644bb4a6ffa50d164bc191c603

screenshot 2016-10-31 16 24 42

**Comment by blowery** [comment]
Tested a bit, looks really good. We might consider altering the way we do the background cover? I tried `50% 17%` and it seemed like it was better at catching faces and things that line up at the top third of the picture.

**Comment by jancavan** [comment]
@blowery Seems to work better at 50% 50%, although it may just be these examples.

**At 50% 50%:**
screenshot 2016-11-01 10 12 07

screenshot 2016-11-01 10 16 17

**At 50% 17%:**
screenshot 2016-11-01 10 14 23

screenshot 2016-11-01 10 15 19

**Comment by fraying** [comment]
Huh. Gotta say, I'm on board with the idea of preferring the upper portion of a portrait oriented photo, but these examples are not convincing. The graffiti frog is better at 17, but the jazz record is better at 50. 

I suggest that we set all of these flexboxes to 50 and leave tweaking like this for later after we live with it for a while. Cool?

**Comment by fraying** [comment]
Otherwise this LGTM!

**Comment by blowery** [comment]
Yeah, leave it 50 and we can play later :)

File: client/blocks/reader-post-card/gallery.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { map, take } from 'lodash';
+
+class PostGallery extends React.Component {
```
Comment (bluefuton) [inline client/blocks/reader-post-card/gallery.jsx]: Yep! That makes a lot of sense. I'll change it.

**Comment by samouri** [comment]
These gallery cards are gorgeous!  The stream is really looking much better with all these new cards 🎉 .  

I noticed the wrapping isn't working well in safari: https://cloudup.com/cO-SkX8TtcW

File: client/blocks/reader-post-card/gallery.jsx (hunk 2 above)
Comment (bluefuton) [inline client/blocks/reader-post-card/gallery.jsx]: Sorted in 714d19c.

**Comment by jancavan** [comment]
In https://github.com/Automattic/wp-calypso/pull/8872/commits/6fca01da0834e2a37cbe038bf8ad15eef19dd20d:

- Fixed cards broken in Safari
- Made breakpoints layout-specific instead of breakpoint-specific

**Comment by jancavan** [comment]
> I noticed the wrapping isn't working well in safari: 

**Before:**
screenshot 2016-11-01 19 08 37

**After:**
screenshot 2016-11-01 19 08 53

**Comment by blowery** [comment]
Seeing something odd on http://calypso.localhost:3000/read/feeds/23920982

maize_n_brew_-_all_posts_ _reader_ _wordpress_com

**Comment by bluefuton** [comment]
> Seeing something odd on http://calypso.localhost:3000/read/feeds/23920982

Looks like malformed Photon image URLs. Here's the second one in React Devtools:

screen shot 2016-11-03 at 15 52 26

The first one appears to work because there's no `filters:no_upscale()` malarkey in there.

I also can't follow that feed (the API thinks it's invalid, but it looks alright to me). I suggest we handle these issues outside this PR - I'll make a P2 post.

**Pull request merged**